### DISCUSSION TITLE
Possibility to compile with PlatformIO?

### DISCUSSION THREAD

**Discussion by alextrical** [Ralim/IronOS#1700] [General]
Is there any possibility for the Firmware to be compiled using Visual Studio Code and PlatformIO
It looks like the base of the FW is FreeRTOS which looks like it may be the same as used by PlatformIO
https://docs.platformio.org/en/stable/frameworks/freertos.html

**Comment by gamelaster** [comment]
It's not about if PlatformIO supports FreeRTOS, rather the SDKs or HALs all around.
Maybe somehow it would be possible ,but why, when you can use CMake extension for VSCode and just use it directly?

**Reply by alextrical** [reply]
I'm more familiar with PlatformIO, and haven't used Cmake for any previous projects. It would potentially allow for more hobbyist users to be able to hop in and add features (think along the lines of the likes of Marlin)

**Reply by Ralim** [reply]
I've used platformio for a few projects.

I've found it tends to try and abstract out too much for this sort of project. I view it as a hard requirement that we have repeatable build conditions; which is why builds are all done in a version locked docker container. Additionally platformio didn't overly seem to handle riscv overly well last I poked it without me doing far more work than just `./start_dev.sh && make`.

Of course if you want to make a PR for this I would be happy to review it. But I don't see it as being a big advantage compared to just using the existing docker build setup.
